src/main/java/com/kth/ciserver/controller/CiController.java

```diff
@@ -25,6 +26,34 @@ public String handleGithubWebhook(@RequestBody GithubWebhookRequest request) {
     }
 
     private void pullAndBuildApplication(GithubWebhookRequest request) {
+        executeAndPrintCommand("git pull");
+        executeAndPrintCommand(String.format("git checkout %s", request.getHead_commit().getId()));
+        Boolean testsSuccessful = executeAndPrintCommand("./gradlew clean build");
+        System.out.println("Tests successful: " + testsSuccessful); // this should instead be printed to file for consistent storage
+    }
 
+    
+    /** 
+     * @param cmd Command to be executed and printed
+     * @return True if the second word in the second to last line of the standard output 
+     * is "SUCCESSFUL" (indicating a successful test run), else false.
+     */
+    private Boolean executeAndPrintCommand(String cmd) {
+        ArrayList<String> lines = new ArrayList<>();
+        try {
+            Runtime run = Runtime.getRuntime();
+            Process process = run.exec(cmd);
+            BufferedReader buf = new BufferedReader(new InputStreamReader(process.getInputStream()));
+            String line;
+            while ((line = buf.readLine()) != null) {
+                System.out.println(line);
+                lines.add(line);
+            }
+        } catch (IOException e) {
+            System.out.println(e.getMessage());
+        }
+        if (lines.size() < 2 || lines.get(lines.size()-2).split(" ").length < 2) 
+            return false;
+        return lines.get(lines.size()-2).split(" ")[1].equals("SUCCESSFUL");
```

Comment:
I'm not sure this is the best way to do this. I quickly looked it up and the "Process" object you use has an exit value https://docs.oracle.com/javase/8/docs/api/java/lang/Process.html#exitValue--

I think we should that here in this return statement, if it isn't 0 then something went wrong.